pkg/app/server/grpcapi/grpcapi.go

```diff
@@ -191,6 +191,22 @@ func getEncriptionKey(se *model.Piped_SecretEncryption) ([]byte, error) {
 	}
 }
 
+func gRPCErrorForEntityOperation(err error, op string, a ...interface{}) error {
```

Comment:
Well, is it really needed to pass `a ...interface{}`?
I feel like it's enough that the caller side just does:
```
gRPCErrorForEntityOperation(err, fmt.Sprintf("add event %s", id))
```

Then:
```
return status.Error(codes.Internal, fmt.Sprintf("Failed to %s", op))
```

`add event 198923jfnk-ao23enfaoe-3fafej` is obviously an operation.

Reply:
I feel you. It is just to help the caller a bit but I have no strong opinions on that. 

I can change that method if you guys want or rename it to be more clear like below. 👍 

```
func gRPCErrorForEntityOperation(err error, opFormat string, opValue ...interface{}) error
```

@khanhtc1202 @knanao Nice to have your thoughts as well.

Reply:
Well, I think this function is just for building the gRPC error code, others should be built outside and pass throw the function parameters. So, how about rename it as
```golang
func gRPCErrorForEntityOperation(err error, msg string) error
```
just like they do for the `status.Error`

![Screen Shot 2022-02-02 at 13 38 16](https://user-images.githubusercontent.com/32532742/152093982-faba1be3-5e6f-4733-80a2-b1e89fa6e207.png)

Reply:
The implementation will be error message wrapper like
```golang
case datastore.ErrNotFound:
	return status.Error(codes.NotFound, fmt.Sprintf("Entity not found: %s", msg))
```

Reply:
Ok. So let me change to being like that. 👍 
FYI, they have `status.Errorf` as well. https://github.com/grpc/grpc-go/blob/v1.44.0/status/status.go#L63

Reply:
oh nice, then 
```golang
func gRPCEntityOperationErrorf(err error, format string, a ...interface{}) error
```
is good to me as well. I still think that this function is only for wrapping error code, but it's up to you 😉

Reply:
Thank you for your comments. Fixed it. PTAL.
https://github.com/pipe-cd/pipecd/pull/3170/commits/b28d93378472b7d039a421309767d024d8e723fe